app/concepts/tickets/use_cases/validate_ticket.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module Tickets
+  module UseCases
+    class ValidateTicket
+      attr_reader :params
+
+      def initialize(params:)
+        @params = params
+      end
+
+      def call
+        check_ticket_date!(received_ticket_id)
+        validate_ticket
+      end
+
+      private
+
+      def check_ticket_date!(received_ticket_id)
+        ticket = Tickets::Repository.new.show(received_ticket_id)
+        raise TicketDateExpired unless DateTime.now < ticket.reservation.seance.date + 1.hour
+      end
+
+      def validate_ticket
+        ticket = Tickets::Repository.new.show(received_ticket_id)
+        raise QrCodeNotValid unless validate_qr_code(ticket) == true
+        raise TicketAlreadyUsed unless ticket.used == false
+
+        update_ticket(ticket.id)
+        true
+      end
+
+      def update_ticket(id)
+        Tickets::UseCases::Update.new.mark_as_used(id: id)
+      end
+
+      def validate_qr_code(ticket)
```

Comment:
```suggestion
      def qr_code_valid?(ticket)
```

I think this is more verbose - `?` suggests that bool is returned, and `validate` is often use for actual validation method - so some error would be expected from `validate` method, not just bool :)